dedupe/manifests/record-linkage.yml

```diff
@@ -0,0 +1,154 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: postgres-db-config
+  labels:
+    app: postgresql-db
+data:
+  POSTGRES_DB: testdb
+  POSTGRES_USER: postgres
+  POSTGRES_PASSWORD: pw
+  PGUSER: postgres
+  PGDATA: /data/pgdata
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgres-db-lb
+spec:
+  selector:
+    app: postgresql-db
+  type: LoadBalancer
+  ports:
+    - port: 5432
+      targetPort: 5432
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: record-linkage-config
+  labels:
+    app: record-linkage
+data:
+  MPI_DB_TYPE: postgres
+  MPI_DBNAME: testdb
+  MPI_PORT: '5432'
+  MPI_USER: postgres
+  MPI_PASSWORD: pw
+  MPI_PATIENT_TABLE: patient
+  MPI_PERSON_TABLE: person
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: record-linkage-service
+spec:
+  selector:
+    app: record-linkage
+  type: LoadBalancer
+  ports:
+    - port: 8080
+      targetPort: 8080
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: postgresql-db
+spec:
+  serviceName: postgresql-db-service
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgresql-db
+  template:
+    metadata:
+      labels:
+        app: postgresql-db
+    spec:
+      containers:
+        - name: postgresql-db
+          image: postgres:latest
+          volumeMounts:
+            - name: postgresql-db-disk
+              mountPath: /data
+          envFrom:
+            - configMapRef:
+                name: postgres-db-config
+          livenessProbe:
+            exec:
+              command:
+                - pg_isready
+            failureThreshold: 20
+            periodSeconds: 2
+            timeoutSeconds: 5
+          ports:
+            - containerPort: 5432
+              name: "http"
+  volumeClaimTemplates:
+    - metadata:
+        name: postgresql-db-disk
+      spec:
+        accessModes: ["ReadWriteOnce"]
+        resources:
+          requests:
+            storage: 25Gi
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: record-linkage
+  labels:
+    app: record-linkage
+spec:
+  selector:
+    matchLabels:
+      app: record-linkage
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: record-linkage
+    spec:
+      initContainers:
+        - name: await-db-ready
```

Comment:
I see a postgres service created on line 69, whats the difference between that one and this

Reply:
This step is definitely a unique one in regards to our "Stateful" type design. Kubernetes will just go and provision everything basically all at the same time. Thats okay and quick for some things but not necessarily for databases that need to do migrations which naturally just take time. Thats where `InitContainers` come in handy. Basically if we didn't use initContainers here the api deployment we fail fast and in a broken state due to health check on when communicating with the db service. So, what this does is pauses the api service from doing anything until the db is done getting ready for it.

Reply:
Another option you could consider...The API container has [pyway](https://github.com/jasondcamp/pyway) installed to run the database migrations, so you could likely use the `pyway validate` command to check to see if the connection is ready in a container readinessProbe.  Outcome should be the same, but you'd have 1 less container.

```sh
pyway --database-type postgres --database-host ${POSTGRES_DB_LB_SERVICE_HOST} --database-port 5432 --database-name ${POSTGRES_DB} --database-username ${POSTGRES_USER} --database-password ${POSTGRE_PASSWORD} --database-migration-dir migrations/ validate
```

Reply:
Oh yeah thats not a bad idea!! i like it. From my understanding with Initcontainers, they are no longer around after they are complete BUT that idea maybe be more efficient in regards to startup. Something to try for sure.

Reply:
I actually got this is work with just the health check endpoint `/` on the `readinessProbe`. Its something I actually tried doing early on into this setup but I must have missed on using delay times or something and never got it to work then. Happy it does now though. I'm about to push it up. If you could please double check next week or something on your end it works that would be awesome :)